### PR TITLE
Callback docs with autosummary

### LINKED ISSUE #2034: Docs are missing the anchor links
## 📚 Documentation

As pointed out by @oplatek the docs suddenly miss the anchor button that allows one to generate a link that points to a particular resource within a page.

This was working before, but now there is a 404 when accessing some assets (js, css, ...)

EDIT: seems not related to the 404 seen in the JS console.

### PR DESCRIPTION
**Pull request by ydcjeff**
## What does this PR do?

Fixes #2034 

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

@williamFalcon @Borda @awaelchli @edenlightning

### CODE REVIEW

File: pytorch_lightning/callbacks/early_stopping.py [hunk 1]
```diff
@@ -40,6 +40,7 @@
 
 class EarlyStopping(Callback):
     r"""
+    Monitor a validation metric and stop training when it stops improving.
```
Comment (ydcjeff) [inline pytorch_lightning/callbacks/early_stopping.py:43]: Adding this to appear the docstring in the left side of table row.

File: docs/source/_static/main.css [hunk 2]
```diff
@@ -0,0 +1,3 @@
+col {
+    width: 50% !important;
+}
```
Comment (ydcjeff) [inline docs/source/_static/main.css:3]: This file is needed as there is `colgroup` html tag which apply width 10% to the right and 90% to the left.
Not sure where does first appear, can't inspect that in pytorch docs.
So manually setting it.

File: docs/source/_templates/classtemplate.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+{{ name | underline }}
+
+.. autoclass:: {{ name }}
+    :members:
+
+
+..
+  autogenerated from source/_templates/classtemplate.rst
+  note it does not have :inherited-members:
```
Comment (ydcjeff) [inline docs/source/_templates/classtemplate.rst:14]: This is the template for `autoclass` to appear like pytorch docs.
without this, the `h1` will be `module.{module}.class`, which is long for sub packages.

**Comment by ydcjeff** [comment]
Shall we allow private methods?

**Comment by ydcjeff** [comment]
Changes here: https://61555-178626720-gh.circle-artifacts.com/0/html/callbacks.html#built-in-callbacks

In ModelCheckpoint or some other like, referencing to other classes, class's params will not show the link as the referencing link for those isn't include in this PR as `api` docs deleted and to keep PR reviewable.

Created `Trainer` docs with autosummary locally and referencing works as before.

In the latest docs, those ref links point to `api` docs.

**Comment by Borda** [comment]
very nice :]
@ydcjeff mind check if we can with this new plugin export methods to be visible/listed on the right side?
basically, it means that the method has to be some kind of heading...

**Comment by Borda** [comment]
@ydcjeff regarding your this docs update, is this is relevant? #3503
because now I see that links to source work fine...

**Comment by ydcjeff** [comment]
> very nice :]
> @ydcjeff mind check if we can with this new plugin export methods to be visible/listed on the right side?
> basically, it means that the method has to be some kind of heading...

Unfortunately, autosummary exports the class with autoclass, function with autofunction

So, appending heading is kinda hard.
I initially thought abt this, but couldn't figure out how to do. will find a way.

**Comment by ydcjeff** [comment]
> @ydcjeff regarding your this docs update, is this is relevant? #3503
> because now I see that links to source work fine...

What the PR does linking the source code to the docs from _modules folder.

For methods like training_step in LightningModule docs can be solved with automethod.

It can be said that PR is relevant or not relevant.

The core team needs to decide source will link to github or link to _modules folder.

**Pull request merged**